pkg/api/rest/create.go

```diff
@@ -68,6 +69,14 @@ func BeforeCreate(strategy RESTCreateStrategy, ctx api.Context, obj runtime.Obje
 	if errs := strategy.Validate(ctx, obj); len(errs) > 0 {
 		return errors.NewInvalid(kind, objectMeta.Name, errs)
 	}
+
+	// Custom validation (including name validation) passed
+	// Now run common validation on object meta
+	// Do this *after* custom validation so that specific error messages are shown whenever possible
+	if errs := validation.ValidateObjectMeta(objectMeta, strategy.NamespaceScoped(), validation.ValidatePathSegmentName); len(errs) > 0 {
```

Comment:
We can be simpler, I think.  Every single object has a name that is DNS1123Subdomain or stricter.  You should just be able to pass pkg/api/validation.go:NameIsDNSSubdomain() here, and get rid of this new impl below...

I'd rather we get more rigorous about validation, but we can get there in the next release.